src/test/scala/io/ino/solrs/CloudSolrServersSpec.scala

```diff
@@ -58,6 +61,48 @@ class CloudSolrServersSpec extends FunSpec with Matchers {
         shardReplica("http://server6:8983/solr/my-collection_shard3_replica2", Enabled, isLeader = false))
     }
 
+    it("should deliver http base url for admin requests") {
+      import scala.collection.JavaConverters._
+
+      val bytes = Files.readAllBytes(Paths.get(this.getClass.getResource("/cluster_status.json").toURI))
+      val cs = ClusterState.load(1, bytes, Set("server1:8983_solr").asJava)
+
+      val collectionToServers = CloudSolrServers.getCollections(cs)
+
+      val cloudServer = new CloudSolrServers[CompletionStage]("zooKeeperHost:2181", defaultCollection = Some("my-collection"))(JavaFutureFactory)
+      cloudServer.collections = collectionToServers
+      cloudServer.nodes = cs.getLiveNodes.asScala.toSeq
+
+      val queryRequest = new GenericSolrRequest(SolrRequest.METHOD.POST, "sampleUrl", new SolrQuery())
+      val adminRequest = CollectionAdminRequest.createAlias("sampleAlias", "sampleCollection")
+
+      cloudServer.matching(queryRequest).get should contain allOf(
+        shardReplica("http://server1:8983/solr/my-collection_shard1_replica1", Enabled, isLeader = true),
+        shardReplica("http://server2:8983/solr/my-collection_shard1_replica2", Enabled, isLeader = false),
+        shardReplica("http://server3:8983/solr/my-collection_shard2_replica1", Enabled, isLeader = true),
+        shardReplica("http://server4:8983/solr/my-collection_shard2_replica2", Enabled, isLeader = false),
+        shardReplica("http://server5:8983/solr/my-collection_shard3_replica1", Enabled, isLeader = true),
+        shardReplica("http://server6:8983/solr/my-collection_shard3_replica2", Enabled, isLeader = false))
+
+      cloudServer.matching(adminRequest).get should contain(SolrServer("http://server1:8983/solr"))
+
+    }
+
+    it("should deliver https base url for admin requests") {
```

Comment:
If solr nodes are registered in zk with http, why should admin requests go to https?
I don't know the problem that should be solved, but I have the feeling that this solution is not the best one.
At first I'd like to better understand the problem...